jsonschema/cli.py

```diff
@@ -252,7 +261,21 @@ def load(_):
                 raise _CannotLoadFile()
         instances = ["<stdin>"]
 
-    validator = arguments["validator"](schema)
+    if arguments["base_uri"] is None:
```

Comment:
I think GitHub somehow ate my comment here, so rewriting it in case you didn't see it -- but, basically I'm hoping the entirety of this change should be `resolver = RefResolver(base_uri=arguments["base_uri"], referrer=schema) if arguments["base_uri"] is not None else None` (or thereabouts).

What goes wrong / what functionality is missing if that's what we have?

Reply:
> I think GitHub somehow ate my comment here, 

Maybe it's a bug of GitHub. Let's ignore it and keep looking at this PR.

>so rewriting it in case you didn't see it -- but, basically I'm hoping the entirety of this change should be `resolver = RefResolver(base_uri=arguments["base_uri"], referrer=schema) if arguments["base_uri"] is not None else None` (or thereabouts).
> 
> What goes wrong / what functionality is missing if that's what we have?

If here is changed to `resolver = RefResolver(base_uri=arguments["base_uri"], referrer=schema) if arguments["base_uri"] is not None else None`, relative paths, such as `--base-uri .`, cannot be supported. If `--base-uri .` and other relative paths are not considered, there will be no problems.